pkg/generator/helper/helper.go

```diff
@@ -188,35 +188,130 @@ func Test{{.CobraCmdFuncName}}(t *testing.T) {
 
 const stepGoImplementationTemplate = `package cmd
 import (
+	"fmt"
 	"github.com/SAP/jenkins-library/pkg/command"
 	"github.com/SAP/jenkins-library/pkg/log"
 	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"github.com/SAP/jenkins-library/pkg/piperutils"
 )
 
+type {{.StepName}}Utils interface {
+	command.ExecRunner
+
+	FileExists(filename string) (bool, error)
+
+	// Add more methods here, or embed additional interfaces, for everything you need to be able to mock in tests.
+	// Unit tests shall be executable in parallel (not depend on global state), and don't (re-)test dependencies.
+}
+
+type {{.StepName}}UtilsBundle struct {
+	*command.Command
+	*piperutils.Files
+
+	// Embed more structs as necessary to implement methods or interfaces you add to {{.StepName}}Utils.
+	// Structs embedded in this way must each have a unique set of methods attached.
+	// If there is no struct which implements the method you need, attach the method to
+	// {{.StepName}}UtilsBundle and forward to the implementation of the dependency.
+}
+
+func new{{.StepName | title}}Utils() {{.StepName}}Utils {
+	utils := {{.StepName}}UtilsBundle{
+		Command: &command.Command{},
+		Files:   &piperutils.Files{},
+	}
+	// Reroute command output to logging framework
+	utils.Stdout(log.Writer())
+	utils.Stderr(log.Writer())
+	return &utils
+}
+
 func {{.StepName}}(config {{ .StepName }}Options, telemetryData *telemetry.CustomData{{ range $notused, $oRes := .OutputResources}}, {{ index $oRes "name" }} *{{ index $oRes "objectname" }}{{ end }}) {
-	// for command execution use Command
-	c := command.Command{}
-	// reroute command output to logging framework
-	c.Stdout(log.Writer())
-	c.Stderr(log.Writer())
+	// Utils can be used wherever the command.ExecRunner interface is expected.
+	// It can also be used for example as a mavenExecRunner.
```

Comment:
You cannot use `command.ExecRunner` in another package, since that introduces circular dependencies when you also include `command.ExecRunner` in the cmd package. Declaring the (minimal) interface at the consumer is Idiomatic Go, see also here https://github.com/golang/go/wiki/CodeReviewComments#interfaces.